### PR TITLE
(google) Add retries around GCS calls.

### PR DESCRIPTION
**Pull request by duftler**

### CODE REVIEW

**Comment by duftler** [comment]
@lwander or @ewiseblatt please review.

**Comment by duftler** [comment]
Depends on https://github.com/spinnaker/spinnaker-dependencies/pull/70.

**Comment by duftler** [comment]
I focused on adding retries to the retrievals so a temporary failure (like a timeout or a network issue) doesn't cause an upstream operation to fail (e.g. a socket timeout while retrieving a child pipeline should not cause the parent pipeline to fail).

File: front50-gcs/src/main/java/com/netflix/spinnaker/front50/config/GcsConfig.java [hunk 1]
```diff
@@ -53,6 +54,18 @@
 @ConditionalOnExpression("${spinnaker.gcs.enabled:false}")
 @EnableConfigurationProperties(GcsProperties.class)
 public class GcsConfig {
+  @Value("${spinnaker.gcs.safeRetryMaxWaitIntervalMs:60000}")
+  Long maxWaitInterval;
+
+  @Value("${spinnaker.gcs.safeRetryRetryIntervalBaseSec:2}")
```
Comment (lwander) [inline front50-gcs/src/main/java/com/netflix/spinnaker/front50/config/GcsConfig.java]: Why not put all of these under a `safeRetry` block? `safeRetryRetry...` reads kind of funny

File: front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/GcsStorageService.java [hunk 2]
```diff
@@ -257,25 +284,32 @@ public boolean supportsVersioning() {
...
-  public <T extends Timestamped> T
-  loadObject(ObjectType objectType, String objectKey)
-         throws NotFoundException {
+  public <T extends Timestamped> T loadObject(ObjectType objectType, String objectKey) throws NotFoundException {
     String path = keyToPath(objectKey, objectType.group);
     try {
-      StorageObject storageObject = timeExecute(loadTimer, obj_api.get(bucketName, path));
-      T item = deserialize(storageObject, (Class<T>) objectType.clazz, true);
-      item.setLastModified(storageObject.getUpdated().getValue());
+      StorageObject[] storageObjectHolder = new StorageObject[1];
```
Comment (lwander) [inline front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/GcsStorageService.java]: What does this accomplish over writing `StorageObject storageObject = null`, and then assigning to that? Oh - does this have to be effectively final?

Comment (duftler) [inline front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/GcsStorageService.java]: Yup.

File: front50-gcs/src/main/java/com/netflix/spinnaker/front50/config/GcsConfig.java (hunk 1 above)
Comment (duftler) [inline front50-gcs/src/main/java/com/netflix/spinnaker/front50/config/GcsConfig.java]: Good point. Done.

File: front50-gcs/front50-gcs.gradle [hunk 3]
```diff
@@ -19,6 +19,7 @@ dependencies {
   compile 'io.reactivex:rxjava:1.0.10'
 
   spinnaker.group('google')
+  compile spinnaker.dependency('clouddriverGoogleCommon')
   compile spinnaker.dependency("korkSecurity")
 
   /* spinnaker.group('googleStorage') */
```
Comment (duftler) [inline front50-gcs/front50-gcs.gradle]: Turns out that dependency `googleStorage` is already in the `google` group, which is included just above. So I've dropped the line entirely.

File: front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/GcsStorageService.java (hunk 2 above)
Comment (lwander) [inline front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/GcsStorageService.java]: gr8

**Pull request merged**